pkg/templates/restartpolicy/template_test.go

```diff
@@ -0,0 +1,118 @@
+package restartpolicy
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/suite"
+	"golang.stackrox.io/kube-linter/pkg/diagnostic"
+	"golang.stackrox.io/kube-linter/pkg/lintcontext/mocks"
+	"golang.stackrox.io/kube-linter/pkg/templates"
+	"golang.stackrox.io/kube-linter/pkg/templates/restartpolicy/internal/params"
+	appsV1 "k8s.io/api/apps/v1"
+	coreV1 "k8s.io/api/core/v1"
+)
+
+func TestRestartPolicy(t *testing.T) {
+	suite.Run(t, new(RestartPolicyTestSuite))
+}
+
+type RestartPolicyTestSuite struct {
+	templates.TemplateTestSuite
+
+	ctx *mocks.MockLintContext
+}
+
+func (s *RestartPolicyTestSuite) SetupTest() {
+	s.Init(templateKey)
+	s.ctx = mocks.NewMockContext()
+}
+
+func (s *RestartPolicyTestSuite) addDeploymentWithRestartPolicy(name string, policy coreV1.RestartPolicy) {
+	s.ctx.AddMockDeployment(s.T(), name)
+	s.ctx.ModifyDeployment(s.T(), name, func(deployment *appsV1.Deployment) {
+		deployment.Spec.Template.Spec.RestartPolicy = policy
+	})
+}
+
+func (s *RestartPolicyTestSuite) addDeploymentWithEmptyRestartPolicy(name string) {
+	s.ctx.AddMockDeployment(s.T(), name)
+	s.ctx.ModifyDeployment(s.T(), name, func(deployment *appsV1.Deployment) {
+		deployment.Spec.Template.Spec.RestartPolicy = ""
+	})
+}
+
+func (s *RestartPolicyTestSuite) addDeploymentWithoutRestartPolicy(name string) {
+	s.ctx.AddMockDeployment(s.T(), name)
+}
+
+func (s *RestartPolicyTestSuite) addObjectWithoutPodSpec(name string) {
+	s.ctx.AddMockService(s.T(), name)
+}
+
+func (s *RestartPolicyTestSuite) TestInvalidRestartPolicies() {
+	const (
+		withoutRestartPolicy = "without-restart-policy"
+		emptyRestartPolicy   = "empty-restart-policy"
+		restartPolicyNever   = "restart-policy-never"
+	)
+
+	s.addDeploymentWithoutRestartPolicy(withoutRestartPolicy)
+	s.addDeploymentWithEmptyRestartPolicy(emptyRestartPolicy)
+	s.addDeploymentWithRestartPolicy(restartPolicyNever, coreV1.RestartPolicyNever)
+
+	s.Validate(s.ctx, []templates.TestCase{
+		{
+			Param: params.Params{},
+			Diagnostics: map[string][]diagnostic.Diagnostic{
+				withoutRestartPolicy: {
+					{Message: "object has a restart policy defined with '' but the only accepted restart policies are '[Always OnFailure]'"},
```

Comment:
Since the default is _Always_, I'm not sure to understand why it's expected to fail when restartPolicy is not defined.

Reply:
On datree documentation, it says that empty value is correct:
https://hub.datree.io/built-in-rules/ensure-valid-restart-policy

Reply:
For the `kube-linter`, the rule is not the same as `datree`; the values accepted by `kube-linter` are `Always` and `OnFailure`.
Any other value is considered invalid, as `Never` or an empty value.